### ISSUE TITLE
[Tune] Tune sklearn tutorial fails on colab

### ISSUE THREAD

**Issue by matthiasmfr** [ray-project/ray#40349]
### What happened + What you expected to happen

The tune sklearn tutorial is not executable, at least not on google colab. 

https://docs.ray.io/en/latest/tune/examples/tune-sklearn.html

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
[<ipython-input-10-91d5d743b03a>](https://localhost:8080/#) in <cell line: 8>()
      6 
      7 start = time.time()
----> 8 tune_search.fit(x_train, y_train)
      9 end = time.time()
     10 print("Tune GridSearch Fit Time:", end - start)

4 frames
[/usr/local/lib/python3.10/dist-packages/ray/tune/analysis/experiment_analysis.py](https://localhost:8080/#) in __init__(self, experiment_checkpoint_path, storage_filesystem, trials, default_metric, default_mode)
    123             if experiment_json_filename is None:
    124                 pattern = TuneController.CKPT_FILE_TMPL.format("*")
--> 125                 raise ValueError(
    126                     f"No experiment checkpoint file of form '{pattern}' was found at: "
    127                     f"({self._fs.type_name}, {self._experiment_fs_path})\n"

ValueError: No experiment checkpoint file of form 'experiment_state-*.json' was found at: (local, /root/ray_results/_Trainable_2023-10-14_16-31-16)
Please check if you specified the correct experiment path, which should be a combination of the `storage_path` and `name` specified in your run.
```

### Versions / Dependencies

Running on google colab
Python version: | 3.10.12
Ray version: | 2.7.1

### Reproduction script
```python
# Keep this here for https://github.com/ray-project/ray/issues/11547
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import train_test_split
from sklearn.linear_model import SGDClassifier
from sklearn.datasets import make_classification
from sklearn import datasets

from ray.tune.sklearn import TuneGridSearchCV
from ray.tune.sklearn import TuneSearchCV

# Create dataset
X, y = make_classification(
    n_samples=11000,
    n_features=1000,
    n_informative=50,
    n_redundant=0,
    n_classes=10,
    class_sep=2.5,
)
x_train, x_test, y_train, y_test = train_test_split(X, y, test_size=1000)

# Example parameters to tune from SGDClassifier
parameter_grid = {"alpha": [1e-4, 1e-1, 1], "epsilon": [0.01, 0.1]}


tune_search = TuneGridSearchCV(
    SGDClassifier(), parameter_grid, early_stopping=True, max_iters=10
)

import time  # Just to compare fit times

start = time.time()
tune_search.fit(x_train, y_train)
end = time.time()
print("Tune GridSearch Fit Time:", end - start)
# Tune GridSearch Fit Time: 15.436315774917603 (for an 8 core laptop)
```
### Issue Severity

Low: It annoys or frustrates me.

**Comment by briancpark** [comment]
I faced the same error when trying to tune a model under Sklearn. Just did a few quick downgrades and seems like it works on 2.6.0, but not on 2.7.0 or after? I don't know how to set and pass `storage_path` and `name` based on reading the documentation.

**Comment by TheDohn** [comment]
I found that setting the local_dir argument of TuneGridSearchCV() will fix this error.

**Comment by justinvyu** [comment]
This has been fixed in `tune-sklearn==0.5.0`. Let me know if that works for you.

**Comment by matthiasmfr** [comment]
Does. Thanks much!
